src/core/ActivityInfoTree.h

```diff
@@ -59,12 +61,17 @@ protected slots:
 private:
     // this is the full activity list, it never changes
     QList<ActivityInfo *> m_menuTreeFull;
+    QList<ActivityInfo *> m_menuTreeFulltemp;
     // represents the Menu view and can be filtered
     QList<ActivityInfo *> m_menuTree;
+    QList<ActivityInfo *> m_searched;
     ActivityInfo *m_rootMenu;
     ActivityInfo *m_currentActivity;
     static int menuTreeCount(QQmlListProperty<ActivityInfo> *property);
     static ActivityInfo *menuTreeAt(QQmlListProperty<ActivityInfo> *property, int index);
+    QObject *m_mainWindow;
+    int templength = 0;
+    QString searchedtext;
```

Comment:
Do we need to optimize it ? According to me yes.
I'll try to explain :
We have a total of more than 100 activities .The first time user types something in the text field the search is performed on those 100 or more activities ,now if we don't use optimization we keep performing the search on those 100 or more activities again and again  which  makes the feature a bit slow.I tested this using "time" library available in c++ to see how much time cpu takes for the computation ,with optimization it takes quite less time compared to a non-optimized code. 

According to what you suggested I have made some changes to the existing code .This is how it looks right now :
I am using m_menutree for filtering but we will still need one extra list(m_searched in this case).

Use  of "m_searched";
if the user changes the tab from search to some other tab without erasing the text  he searched then we need to display the activities   that were searched. Since the value of m_menutree gets changed on changing the tab  we need a temp list which will store the activities that were  searched 

void display(QString text)
{
       int length,start = 0 ;

    length = text.length();
    searchedtext = text;
    m_menuTree = m_searched;


    if(length < templength)
    {
        m_menuTree = m_menuTreeFull;

    }
    if(text.trimmed().isEmpty())
    {
        m_menuTree = m_menuTreeFull;
        templength = 0 ;

    }
    else
    {

        if(length != templength)
        {
            qDebug()<<endl<<"hey";
            while(start != m_menuTree.length())
            {
                if(!m_menuTree.at(start)->title().contains(text.trimmed(),Qt::CaseInsensitive))
                {
                    m_menuTree.removeAt(start);
                    start = start - 1;

                }
                start ++;

            }
        }
        templength = length;

    }
    m_searched = m_menuTree;
    filterEnabledActivities();
    filterLockedActivities();
    sortByDifficulty();
    emit menuTreeChanged();

}